src/components/VisualizationStep.tsx

```diff
@@ -207,53 +208,16 @@ const VisualizationStep = ({ data }: NodeProps<IVizStepNodeData>) => {
           )}
 
           {/* ADD/APPEND STEP BUTTON */}
-          {VisualizationService.showAppendStepButton(data, endStep) ? (
-            <Popover
-              id="popover-append-step"
-              appendTo={() => document.body}
-              aria-label="Add a step or branch"
-              bodyContent={
-                <MiniCatalog
-                  children={<BranchBuilder handleAddBranch={handleAddBranch} />}
-                  disableBranchesTab={!showBranchesTab}
-                  disableBranchesTabMsg={ValidationService.getBranchTabTooltipMsg(
-                    supportsBranching,
-                    data.step.maxBranches,
-                    data.step.branches?.length
-                  )}
-                  disableStepsTab={!showStepsTab}
-                  disableStepsTabMsg={"You can't add a step between a step and a branch."}
-                  handleSelectStep={onMiniCatalogClickAppend}
-                  queryParams={{
-                    dsl: currentDSL,
-                    type: ValidationService.appendableStepTypes(data.step.type),
-                  }}
-                  step={data.step}
-                />
-              }
-              className={'miniCatalog__popover'}
-              data-testid={'miniCatalog__popover'}
-              enableFlip={true}
-              flipBehavior={['top-start', 'left-start']}
-              hasAutoWidth
-              hideOnOutsideClick={true}
-              position={'right-start'}
-              showClose={false}
-            >
-              <Tooltip
-                content={ValidationService.getPlusButtonTooltipMsg(showBranchesTab, showStepsTab)}
-              >
-                <button
-                  className="stepNode__Add plusButton nodrag"
-                  data-testid={'stepNode__appendStep-btn'}
-                >
-                  <PlusIcon />
-                </button>
-              </Tooltip>
-            </Popover>
-          ) : (
-            <></>
-          )}
+          {VisualizationService.showAppendStepButton(data, endStep)
+            && <AppendStepButton
+              handleAddBranch={handleAddBranch}
+              handleSelectStep={onMiniCatalogClickAppend}
+              step={data.step}
+              showBranchesTab={showBranchesTab}
+              showStepsTab={showStepsTab}
+              supportsBranching={supportsBranching}
+            />
+          }
```

Comment:
This is the initial step, IMHO, the next candidate to move into its own component should be the `Add step` placeholder.

Reply:
I'm very happy with this, as I've been meaning to get some of these components moved into smaller ones for a while! FYI, you can also just write `showStepsTab` instead of `showStepsTab={showStepsTab}`. I don't mind either way, but just wanted to share in case you didn't already know!

Reply:
> but just wanted to share in case you didn't already know!

@kahboom thanks for sharing, I didn't know, but just to clarify, wouldn't that force to `showStepsTab` property to be true always?